src/main/kotlin/com/sourcegraph/cody/config/SettingsMigration.kt

```diff
@@ -118,6 +129,54 @@ class SettingsMigration : Activity {
         }
   }
 
+  fun migrateDeprecatedChatLlms(project: Project) {
+    CodyAgentService.withAgent(project) { agent ->
+      val chatModels = agent.server.chatModels(ChatModelsParams(ModelUsage.CHAT.value))
+      val models =
+          chatModels.completeOnTimeout(null, 10, TimeUnit.SECONDS).get()?.models ?: return@withAgent
+      val defaultModel = models.find { it.default } ?: return@withAgent
+
+      fun isDeprecated(modelState: LLMState): Boolean =
+          models.firstOrNull { it.model == modelState.model }?.deprecated ?: false
+
+      val migratedLlms = mutableSetOf<String>()
+      HistoryService.getInstance(project).state.accountData.forEach { accountData ->
+        accountData.chats
+            .mapNotNull { it.llm }
+            .forEach { llm ->
+              if (isDeprecated(llm)) {
+                llm.title?.let { migratedLlms.add(it) }
+                llm.model = defaultModel.model
+                llm.title = defaultModel.title
+                llm.provider = defaultModel.provider
+              }
+            }
+      }
+
+      if (migratedLlms.isNotEmpty()) {
+        ApplicationManager.getApplication().invokeLater {
+          val msg =
+              """
+              <html>
+              All chats have been upgraded to newer, more capable models. Some old models are no longer supported:<br>
```

Comment:
can you please bundle this message? also, maybe it would be cleaner if you extract this notification to the separate class?

Reply:
I can extract it to a bundle, but I'm not sure if it makes sense to extract to a separate class if its used only in this one place.

Reply:
Hm, or maybe you mean we could extract whole migration to a separate class?
I can do that.